#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/robots/universal_robots/protos/UR10e.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/tables/protos/Table.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/factory/containers/protos/MetalStorageBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/devices/robotiq/protos/Robotiq3fGripper.proto"
EXTERNPROTO "../protos/PiecePrismatique.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.12182768154631633 0.7863865632678861 0.6056023357953404 0.5008994491555835
  position -4.526501284417437 -0.4389157404266282 3.007108144694703
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.02 0 0
  floorSize 10 10
}
Robot {
  translation -0.4 -0.14 0
  children [
    Track {
      translation -6.679601316506023e-12 -9.672568301866136e-12 0.4498195586644608
      rotation 0.026746348040125635 0.9996422524415963 9.127625156805124e-08 4.404265978471752e-05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/rubber/dotted_rubber_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/rubber/dotted_rubber_roughness.jpg"
              ]
            }
            metalness 0
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/rubber/dotted_rubber_normal.jpg"
              ]
            }
            occlusionMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/rubber/dotted_rubber_occlusion.jpg"
              ]
            }
            textureTransform TextureTransform {
              scale 12 2
              translation -0.241066666666666 0
            }
          }
          geometry DEF BELT_BO Box {
            size 3 0.5 0.3
          }
        }
      ]
      boundingObject USE BELT_BO
      physics Physics {
        density -1
        mass 1
      }
      device [
        LinearMotor {
          name "belt_motor"
          maxVelocity 0.5
          sound ""
        }
      ]
      textureAnimation 0.3333333333333333 0
      linearVelocity 1.95692991328567e-11 2.824756400499777e-11 3.5042489798811557e-10
      angularVelocity 5.548350891590269e-12 -1.1515894416805021e-10 -1.2278413662888979e-11
    }
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/corrugated_metal/corrugated_metal_base_color.jpg"
          ]
        }
        roughnessMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/corrugated_metal/corrugated_metal_roughness.jpg"
          ]
        }
        metalnessMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/corrugated_metal/corrugated_metal_metalness.jpg"
          ]
        }
        normalMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/corrugated_metal/corrugated_metal_normal.jpg"
          ]
        }
        occlusionMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/textures/corrugated_metal/corrugated_metal_occlusion.jpg"
          ]
        }
        textureTransform TextureTransform {
          scale 2 2
        }
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            1.53 -0.28 0
            1.53 -0.28 0.61
            -1.53 -0.28 0.61
            -1.53 -0.28 0
            1.53 0.28 0
            1.53 0.28 0.61
            -1.53 0.28 0.61
            -1.53 0.28 0
            1.5 -0.25 0.61
            1.5 0.25 0.61
            -1.5 0.25 0.61
            -1.5 -0.25 0.61
            1.5 -0.25 0.6
            1.5 0.25 0.6
            -1.5 0.25 0.6
            -1.5 -0.25 0.6
          ]
        }
        texCoord TextureCoordinate {
          point [
            0 0
            1 0
            1 1
            0 1
            0 0.3986928104575163
            1 0.3986928104575163
            0 0.3986928104575163
            0.18300653594771243 0.3986928104575163
            0.18300653594771243 0
            0 0
            0.001 0
            0.001 0.001
            0 0.001
          ]
        }
        coordIndex [
          0, 1, 2, 3, -1, 4, 7, 6, 5, -1, 0, 4, 5, 1, -1, 7, 3, 2, 6, -1, 1, 5, 9, 8, -1, 5, 6, 10, 9, -1, 6, 2, 11, 10, -1, 2, 1, 8, 11, -1, 0, 3, 7, 4, -1, 8, 9, 13, 12, -1, 9, 10, 14, 13, -1, 10, 11, 15, 14, -1, 11, 8, 12, 15, -1
        ]
        texCoordIndex [
          1, 5, 4, 0, -1, 0, 1, 5, 4, -1, 0, 8, 7, 6, -1, 0, 8, 7, 6, -1, 9, 10, 11, 12, -1, 9, 10, 11, 12, -1, 9, 10, 11, 12, -1, 9, 10, 11, 12, -1, 9, 10, 11, 12, -1, 9, 10, 11, 12, -1, 9, 10, 11, 12, -1, 9, 10, 11, 12, -1, 9, 10, 11, 12, -1
        ]
      }
    }
    DistanceSensor {
      translation 0.659952 0.270272 0.63
      rotation 0 0 1 -1.57
      name "ds1"
      lookupTable [
        0 0 0
        0.5 1000 0
      ]
    }
    Emitter {
      channel 2
    }
    Emitter {
      name "arm_emitter"
      channel 5
    }
    DistanceSensor {
      translation 0.849952 0.270424 0.63
      rotation 0 0 1 -1.57
      name "ds2"
      lookupTable [
        0 0 0
        0.5 1000 0
      ]
    }
    DistanceSensor {
      translation 1.15995 0.270672 0.62
      rotation 0 0 1 -1.57
      name "ds3"
      lookupTable [
        0 0 0
        0.5 1000 0
      ]
    }
    Receiver {
      channel 3
    }
    Receiver {
      name "arm_receiver"
      channel 3
    }
  ]
  name "conveyor belt(1)"
  boundingObject Group {
    children [
      Pose {
        translation 1.515 0 0.305
        children [
          Box {
            size 0.03 0.56 0.61
          }
        ]
      }
      Pose {
        translation -1.515 0 0.305
        children [
          Box {
            size 0.03 0.56 0.61
          }
        ]
      }
      Pose {
        translation 0 -0.265 0.305
        children [
          Box {
            size 3 0.03 0.61
          }
        ]
      }
      Pose {
        translation 0 0.265 0.305
        children [
          Box {
            size 3 0.03 0.61
          }
        ]
      }
    ]
  }
  controller "conveyor_controller"
  controllerArgs [
    "0.2"
    "0"
  ]
  window "<none>"
}
PiecePrismatique {
  translation -0.0193466 -0.0351611 0.66
  rotation -0.2592038987710034 -0.9658226228094736 -1.1460395524278974e-05 -3.141565307179586
  color 0.333333 0 0
}
PiecePrismatique {
  translation -0.850231 -0.28932 0.65
  rotation -0.966029666749716 0.00038376686761232905 0.2584309108493482 1.57059
  color 0.333333 0 0
}
PiecePrismatique {
  translation -1.45011 -0.118844 0.65
  rotation 1.4573599998299013e-05 -4.587399999464572e-06 0.9999999998832829 -1.0480053071795865
  color 0.333333 0 0
}
Table {
  translation 3.67318e-08 -1.21 0
  rotation 0 0 1 1.5708
  size 1 1.8 0.63
}
MetalStorageBox {
  translation 0.93 -2.33 0
}
UR10e {
  hidden position_0_0 -0.779599999788954
  hidden rotation_1 0 0 -1 0.7795999997889538
  hidden position_1_0 -0.5068646756830736
  hidden linearVelocity_1 -2.3510628318214597e-08 -3.2344875822507054e-10 1.474225930034972e-09
  hidden angularVelocity_1 1.8573978502411028e-09 -6.587627606985834e-10 2.883847969061607e-08
  hidden rotation_2 0 1 0 1.063931324316848
  hidden position_2_0 0.5072440466441749
  hidden linearVelocity_2 -6.272614874528823e-08 -6.361698658023006e-09 3.831605845740289e-09
  hidden angularVelocity_2 3.72980364384044e-09 -1.5778645780072618e-09 5.762652686937145e-08
  hidden rotation_3 0 1 0 0.507244046644175
  hidden position_3_0 -1.569994464063102
  hidden linearVelocity_3 -1.022671852833753e-07 5.434276841865006e-10 6.5421417092873346e-09
  hidden angularVelocity_3 4.70994500501251e-09 4.7460606677258984e-09 7.37553895987932e-08
  hidden rotation_4 0 1 0 0.0008015358943764378
  hidden position_4_0 -1.57000197991349
  hidden linearVelocity_4 -1.688143134031433e-07 -9.045130834144827e-09 -5.643171009848759e-11
  hidden angularVelocity_4 -1.7296296038257623e-06 1.7072451517874062e-09 7.607463724197142e-08
  hidden rotation_5 0 0 -1 1.57000197991349
  hidden position_5_0 5.3075673928161257e-08
  hidden linearVelocity_5 1.0409082615187643e-06 -1.6427209852079638e-08 -1.929413064997311e-07
  hidden angularVelocity_5 -7.727084586841156e-07 -0.00017748487699327197 3.2645694118537797e-07
  hidden rotation_6 0 1 0 5.16191365590357e-08
  hidden linearVelocity_6 1.7358101352951245e-05 -8.796132731885795e-08 -2.0478473229545234e-07
  hidden angularVelocity_6 -7.723904475122421e-07 -0.0001774781829487263 8.721348933189054e-07
  hidden position_11_0 4.1687968040851984e-06
  hidden position_11_1 -1.6734957230864865e-05
  hidden position_11_2 0.0495001927066631
  hidden rotation_12 0.7068252250470913 0.7073882252463052 4.4733981342860215e-06 3.1416009491403423
  hidden position_12_0 0.04951767499319588
  hidden linearVelocity_12 3.193540484062396e-05 9.482002951699542e-08 -8.510456085346619e-06
  hidden angularVelocity_12 -4.3540364233704795e-05 -0.00017770716875711512 9.227955539303585e-07
  hidden rotation_13 0 0 1 1.7674988339800856e-05
  hidden position_13_0 2.8621410037973976e-08
  hidden linearVelocity_13 3.548379753435007e-05 -4.897190314779374e-07 -8.672823420142898e-06
  hidden angularVelocity_13 -4.346818838727251e-05 -0.00017770151132597638 9.081837580244655e-07
  hidden rotation_14 0 0 -1 0.519999971378341
  hidden position_14_0 -0.05230000000002916
  hidden linearVelocity_14 4.4358635200757354e-05 -2.6866369934401256e-06 -1.3651274899757482e-05
  hidden angularVelocity_14 -4.346818604016314e-05 -0.00017770147680610504 9.081826715439234e-07
  hidden linearVelocity_15 5.0367046376731395e-05 -4.174130653748012e-06 -1.7096971479233504e-05
  hidden angularVelocity_15 -4.346818592535519e-05 -0.0001777014768037943 9.081826688507333e-07
  hidden rotation_16 0.7068252251011821 0.7073882252003986 -2.914345199715887e-06 3.1415861618907024
  hidden position_16_0 0.04951767333954216
  hidden linearVelocity_16 3.1983689113985745e-05 -5.45718647231297e-07 -8.385092027722863e-06
  hidden angularVelocity_16 4.978451524949942e-05 -0.00017720351853394909 8.233164747852853e-07
  hidden rotation_17 0 0 1 1.7673179226327022e-05
  hidden position_17_0 2.862378613258621e-08
  hidden linearVelocity_17 3.552546246282895e-05 9.527178486715114e-08 -8.549217999652934e-06
  hidden angularVelocity_17 4.965165652510861e-05 -0.00017720828350876224 8.517299101086791e-07
  hidden rotation_18 0 0 -1 0.519999971375965
  hidden position_18_0 -0.052300000000015716
  hidden linearVelocity_18 4.4375648632467386e-05 2.5515266326826383e-06 -1.3534451701247008e-05
  hidden angularVelocity_18 4.965165239994965e-05 -0.00017720825271651545 8.517316527140004e-07
  hidden linearVelocity_19 5.0367373262663055e-05 4.213869309658627e-06 -1.6984743785251056e-05
  hidden angularVelocity_19 4.9651652193561326e-05 -0.00017720825224750342 8.517316502104615e-07
  hidden rotation_20 0 0 1 1.7377590819732256e-07
  hidden position_20_0 2.8424019312251516e-08
  hidden linearVelocity_20 3.552044967340322e-05 -1.2742976852253007e-07 7.85877336677269e-06
  hidden angularVelocity_20 -7.723884631820018e-07 -0.0001774785539742162 8.721354986208491e-07
  hidden rotation_21 0 0 -1 0.5199999715757581
  hidden position_21_0 -0.05230000000012117
  hidden linearVelocity_21 4.439108161382568e-05 -1.4164821844106064e-07 1.2827841725264004e-05
  hidden angularVelocity_21 -7.723883076726924e-07 -0.0001774785854046428 8.721355426722264e-07
  hidden linearVelocity_22 5.0396754037080886e-05 -1.5089254615139117e-07 1.6267184741600715e-05
  hidden angularVelocity_22 -7.72388309721267e-07 -0.00017747858503098935 8.721355395888878e-07
  translation 0.850066 -1.07005 0.64
  rotation 0 0 -1 -2.3555953071795863
  controller "arm_controller"
  controllerArgs [
    ""
  ]
  toolSlot [
    Receiver {
      channel 1
    }
    Receiver {
      name "conv_receiver"
      channel 5
    }
    Emitter {
      channel 3
    }
    DistanceSensor {
      translation -3.6732e-07 0.1 0
      rotation 0 0 1 1.5708
    }
    Robotiq3fGripper {
    }
    Pose {
      translation -0.029995 0.12 -0.27
      rotation 0 1 0 3.14159
      children [
        Robot {
          translation -1.7084e-07 1.01558e-07 -0.02
          rotation 0.5773489358550934 0.5773499358549823 -0.5773519358547601 -2.0943853071795866
          children [
            Camera {
              rotation 1 0 0 -1.5708
              fieldOfView 1.2
              width 1024
              height 768
              recognition Recognition {
                maxObjects 1
                frameThickness 2
              }
            }
            Receiver {
              channel 2
            }
            Emitter {
              channel 1
            }
          ]
          name "camera_robot"
          controller "camera_controller"
        }
      ]
    }
  ]
}
